docs/src/migration.md

```diff
@@ -0,0 +1,275 @@
+mg
+```
+
+***
+
+###### Create a directed metagraph based on the simplegraph, with optional default edge weight
+
+Using *MetaGraphs.jl*:
+```@example main
+_mdg = MetaGraphs.MetaDiGraph(path_graph(5), 3.0)
+```
+
+Using *MetaGraphsNext.jl*:
+```@example main
+mg = MetaGraphsNext.MetaGraph(DiGraph(), default_weight = 3.0)
+
+# A more concise way to construct a path graph with 5 vertices
+foreach(x -> add_vertex!(mg, Symbol(x), nothing), 'a':'e')
+foreach(x -> add_edge!(mg, Symbol(x[1]), Symbol(x[2]), nothing), zip('a':'e', 'b':'e'))
+mg
+```
+
+***
+
+###### Set some properties for the graph itself
+
+Using *MetaGraphs.jl*:
+```@example main
+MetaGraphs.set_prop!(_mg, :description, "This is a metagraph.")
+```
+
+Using *MetaGraphsNext.jl*:
+```@example main
+# Important: graph data can only be set when constructing the object
+mg = MetaGraphsNext.MetaGraph(DiGraph(), graph_data = "graph_of_colors")
+
+foreach(x -> add_vertex!(mg, Symbol(x), nothing), 'a':'e')
+foreach(x -> add_edge!(mg, Symbol(x[1]), Symbol(x[2]), nothing), zip('a':'e', 'b':'e'))
+```
+
+***
+
+###### Set properties on a vertex in bulk
+
+Using *MetaGraphs.jl*:
+```@example main
+MetaGraphs.set_props!(_mg, 1, Dict(:name=>"Susan", :id => 123))
+```
+
+Using *MetaGraphsNext.jl*:
+```@example main
+mg = MetaGraphsNext.MetaGraph(Graph(),
+                              VertexData = NamedTuple{(:name, :id), Tuple{String, Union{Int64, Missing}}})
+
+foreach(x -> add_vertex!(mg, Symbol(x), (name = "", id = missing)), 'a':'e')
+foreach(x -> add_edge!(mg, Symbol(x[1]), Symbol(x[2]), nothing), zip('a':'e', 'b':'e'))
+
+mg[MetaGraphsNext.label_for(mg, 1)] = (name = "Susan", id = 123)
+```
+
+***
+
+###### Set individual properties
+
+Using *MetaGraphs.jl*:
+```@example main
+MetaGraphs.set_prop!(_mg, 2, :name, "John")
+```
+
+Using *MetaGraphsNext.jl*:
+```@example main
+# Not possible since but this is a workaround
+mg[Symbol(2)] = (name = "John", id = missing)
+```
+
+***
+
+###### Set a property on an edge
+
+Using *MetaGraphs.jl*:
+```@example main
+MetaGraphs.set_prop!(_mg, Edge(1, 2), :action, "knows")
+```
+
+Using *MetaGraphsNext.jl*:
+```@example main
+mg = MetaGraphsNext.MetaGraph(Graph(), 
+                              VertexData = NamedTuple{(:name, :id), Tuple{String, Union{Int64, Missing}}},
+                              EdgeData = String)
+
+foreach(x -> add_vertex!(mg, Symbol(x), (name = "", id = missing)), 1:5)
+foreach(x -> add_edge!(mg, Symbol(x[1]), Symbol(x[2]), ""), zip(1:5, 2:5))
+mg[Symbol(1)] = (name = "Susan", id = 123)
+mg[Symbol(2)] = (name = "John", id = missing)
+
+mg[Symbol(1), Symbol(2)] = "knows"
+```
+
+***
+
+###### Set another property on an edge by specifying source and destination
+
+Using *MetaGraphs.jl*:
+```@example main
+using Dates: Date
+MetaGraphs.set_prop!(_mg, 1, 2, :since, Date("20170501", "yyyymmdd"))
+```
+
+Using *MetaGraphsNext.jl*:
+```@example main
+# Not supported. The exact data type of the vertex data needs to be defined
```

Comment:
You could have a dict with eltype Any as vertex metadata. Not sure that would be advisable, but it would work.

Reply:
I think in general, using dicts with eltype Any as vertex metadata might be the best way to show people how to migrate from MetaGraphs to MetaGraphsNext. Then, afterwards, you could talk about using something more strongly typed, like a NamedTuple or mutable struct, might be better if all the metadata has same structure.